Makefile

```diff
@@ -1,5 +1,17 @@
 GO := go
 GOFMT := gofmt
+PROTOC_INSTALLED := $(shell which protoc 2>/dev/null)
+PROTOC_EXPECTED_VERSION =libprotoc 3.21.12
+PROTOC_CURR_VERSION := $(shell protoc --version 2>/dev/null)
+BUF_INSTALLED := $(shell which buf 2>/dev/null)
+BUF_EXPECTED_VERSION = 1.15.0
```

Comment:
Do we really need this. I thinking I was using something like 1.12.5 at one point and it was fine. Not sure it's that sensitive. Same comment for PROTOC_EXPECTED_VERSION.

I would be more worried about the plugin versions, but pinning those might also cause more hassle than leaving them.

Reply:
the plugin versions should be pinned via tools.go if I understood https://github.com/grpc-ecosystem/grpc-gateway#compile-from-source and the link to https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module correctly, that's why I didn't add sth. around them

I'd prefer to leave the tooling versioncheck in, as I wanna be as sure as I can that we get deterministic and expectable / reproducible output on all machines, and I don't expect this to update very often.

Reply:
If we're worried about this, then it pushes us towards docker again.

Reply:
true, docker is the preferred solution, thought we already discussed that.

Don't get me wrong, I am not hard on this, I am just insecure and want reproducible output, so I added these pinnings as safety nets. Also interested about other opinions, cc @akoserwal @Rajagopalan-Ranganathan @alechenninger @wscalf.

Reply:
ftr: just tested with buf 1.14 and protoc 3.18.0 with same outcome, so i'll remove the nets now as I am sufficiently relieved sofar that we shouldn't care too much.